plugins/SitesManager/API.php

```diff
@@ -959,13 +959,9 @@ public function deleteSite(
 
     private function checkValidTimezone($timezone)
     {
-        $timezones = $this->getTimezonesList();
-        foreach (array_values($timezones) as $cities) {
-            foreach ($cities as $timezoneId => $city) {
-                if ($timezoneId == $timezone) {
-                    return true;
-                }
-            }
+        $timezones = DateTimeZone::listIdentifiers(DateTimeZone::ALL_WITH_BC);
```

Comment:
Looking at the test failures I guess we still need to check for the custom `getTimezonesListUTCOffsets()` entries.

And maybe add a test to ensure this does not regress? Something simple like adding a site with any bc timezone should be good enough:

```php
$timezone = current(array_diff(
    \DateTimeZone::listIdentifiers(\DateTimeZone::ALL_WITH_BC),
    \DateTimeZone::listIdentifiers(\DateTimeZone::ALL)
));

$idSite = API::getInstance()->addSite(
    "site1",
    ['http://example.org'],
    $ecommerce = 0,
    $siteSearch = 1,
    $searchKeywordParameters = null,
    $searchCategoryParameters = null,
    $ip = '',
    $params = '',
    $timezone
);

$site = new Site($idSite);

self::assertSame($timezone, $site->getTimezone());
```